### ISSUE TITLE
[BUG] Unit tests all are relative to executing time-zone. Only works if local machine running is UTC+3

### ISSUE THREAD

**Issue by xucito** [wavesplatform/data-service#305]
**Describe the bug**

When running the test in the project, the tests in sql.test.js that involve date time require the runner to be in UTC+3 to be successful.

**To Reproduce**
1. Set timezone to UTC+4
2. Run jest tests on sql.test.js
3. Observe that the timestamps expected are +3

![image](https://user-images.githubusercontent.com/36687889/73593335-1d369400-4514-11ea-8d25-f7cb7c13a983.png)

**Expected behavior**

Regardless of what time-zone the tests are run in, the outcome should be the same

**Additional context**
Running using vscode Jest Runner

**Comment by xucito** [comment]
To add onto this, I think the generated sql statements are not time-zone safe.
The insert statement uses timestamps such as `1970-01-01 03:00:00.000` in the insert.

I guess this would imply that the sql server and the data-service server must be in the same time-zone or there maybe some time-zone issues.

An example, if the sql server is +3 UTC and the data-service is +4 UTC and you try to save a record with timestamp  2000-01-01 00:00:00.000 + 0, this will result in a record being inserted using the value 
'2000-01-01 04:00:00.000'

However the sql server will think it is  2000 04:00:00.000 relative to its +3 UTC timezone so the real UTC time it saves is 1999-12-31 11:00:00.000 + 0

**Comment by dvshur** [comment]
Hi, @xucito, thank you for the report!

While I agree that unit tests should be fixed, which we'll do at some point, I can assure you that the observed behaviour does not affect production code and logic. Reasons:

1. All blockchain data in a database is received from blockchain and put into database without any timestamp conversion, so, it's in UTC.
2. Our APIs take dates from client requests and pass them along unchanged. So results are consistent with the objects in the blockchain the user searches for — they are, again, in UTC.
3. Daemons that calculate candles and pairs use database timestamps (UTC).

So we haven't experienced any production issues with that behaviour, and don't expect to. Additionally, we run all our servers in Docker, and its containers use UTC by default (I know we shouldn't rely on this, but it is a convenient status quo).

To sum up: I've put a task in the backlog to fix tests, but I don't think it warrants a `Bug` status right now, because there are no real problems for end users.

**Comment by xucito** [comment]
Thanks for the response @dvshur, a solution is to run everything in UTC+0 sure however I think that if you expect everything to be run in one timezone, maybe you can list it somewhere in your documentation.

I would add that this can affect production environments especially as in your recommendations you mention:
"Use several machines in different data centers and a balancer to minimize downtime"

I believe that if your data-service server is running in non-UTC+0 timezone it will still save the time-stamp as UTC+0 so 3:00 AM +3 will be saved as 3:00. The data-service will then read that back up as UTC as it is a date-time without timestamp stored in postgres?

Please correct me if I am wrong but this means that this datetime issue will happen in ANY deployment where the data-service is not running in UTC-0.

**Comment by dvshur** [comment]
> data-service server is running in non-UTC+0 timezone it will still save the time-stamp as UTC+0 so 3:00 AM

The data service itself is a read-only server. It doesn't save anything to the database, it just reads whatever date is stored there, which is UTC. Also, it parses whatever date the client provides on request, and clients use UTC because all objects in the blockchain do as well.

To fill the database we use the crawler (blockchain-postgres-sync), which does not parse dates at all, it just passes data directly from blockchain to postgres.

So no, it shouldn't really matter which timezone your servers are running in, it should work anywhere.

Have you actually tried to run the data service with node on live data, make some live API requests 
on a server with a different timezone and reproduce the bug you're talking about?

**Comment by dvshur** [comment]
I believe I have addressed the question, so closing for now. Please feel free to reopen if you have any more comments.

**Comment by xucito** [comment]
FYI, I was able to replicate this issue.

If you run the pgsql server on a UTC+2 timezone and run the data-service in a UTC+0 timezone, every single candle increment (5m, 15m, 30m etc), will increment the candle time by 2 hours.
